src/engine/InputManager.java

```diff
@@ -71,6 +79,20 @@ public void keyReleased(final KeyEvent key) {
 			keys[key.getKeyCode()] = false;
 	}
 
+	/**
+	 * Checks if a given key code has been double-tapped within the defined threshold.
+	 *
+	 * @param keyCode The key code to check for a double-tap.
+	 * @return True if the key was double-tapped, false otherwise.
+	 */
+	public boolean isDoubleTap(int keyCode) {
+		long currentTime = System.currentTimeMillis();
+		long lastPressTime = lastKeyPress.getOrDefault(keyCode, 0L);
+		boolean isDoubleTap = (currentTime - lastPressTime) <= DOUBLE_TAP_THRESHOLD;
+		lastKeyPress.put(keyCode, currentTime);
```

Comment:
lastPressTime should be updated at the start of input, not during pressing a key.
It always detect double tap when just pressing key